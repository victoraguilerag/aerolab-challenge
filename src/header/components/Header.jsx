import React from 'react'
import PropTypes from 'prop-types'
import logo from '../../aerolab-logo.svg'
import coin from '../../icons/coin.svg'

function Header (props) {
  const { handleModal } = props
  const { user } = props
  const name = user.get('name')
  const points = user.get('points')
  return (
    <div className="header" style={styles.header}>
      <img
        className="logo"
        style={styles.logo}
        src={logo}
        width="39px"
        height="36px"
        alt="brand logo"
      />
      <div className="user-data" style={styles.userData}>
        <p className="user-name" style={styles.userName} onClick={() => handleModal('history')}>{name}</p>
        <div className="user-points" style={styles.userPoints}>
          <p className="points" style={styles.points}>{points}</p>
          <img
            className="coin"
            style={styles.coin}
            width="24px"
            height="24px"
            src={coin}
            alt="coin"
          />
        </div>
      </div>
    </div>
  )
}

const styles = {
  'header': {
    'display': 'flex',
    'flexFlow': 'row nowrap',
    'justifyContent': 'space-between',
    'alignItems': 'center',
    'background': '#ffffff',
    'height': '80px',
    'padding': '0 20px'
  },
  'userData': {
    'minWidth': '220px',
    'display': 'flex',
    'justifyContent': 'space-between',
    'alignItems': 'center'
  },
  'userPoints': {
    'userSelect': 'none',
    'display': 'flex',
    'justifyContent': 'center',
    'alignItems': 'center',
    'background': '#ededed',
    'borderRadius': '100px',
    'width': '103px',
    'height': '48px'
  },
  'userName': {
    'userSelect': 'none',
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left',
    'marginRight': '19px',
    'cursor': 'pointer'
  },
  'points': {
    'userSelect': 'none',
    'fontSize': '24px',
    'color': '#616161',
    'letterSpacing': '-0.15px',
    'lineHeight': '48px',
    'textAlign': 'left',
    'cursor': 'pointer'
  }
}

Header.propTypes = {
  handleModal: PropTypes.func.isRequired,
  user: PropTypes.shape({}).isRequired
}

export default Header
