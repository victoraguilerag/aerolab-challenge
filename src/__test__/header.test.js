import React from 'react'
import Header from '../header/components/Header.jsx'
import renderer from 'react-test-renderer'

const functionMock = () => console.log('mock')
const user = {
  name: 'Patricio Estrella',
  points: 2000
}

test('Header renders', () => {
  const component = renderer.create(
    <Header
      handleModal={functionMock}
      user={user}
    />,
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
