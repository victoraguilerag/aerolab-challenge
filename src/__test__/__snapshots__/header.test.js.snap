// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="header"
  style={
    Object {
      "alignItems": "center",
      "background": "#ffffff",
      "display": "flex",
      "flexFlow": "row nowrap",
      "height": "80px",
      "justifyContent": "space-between",
      "padding": "0 20px",
    }
  }
>
  <img
    alt="brand logo"
    className="logo"
    height="36px"
    src={Object {}}
    width="39px"
  />
  <div
    className="user-data"
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "space-between",
        "minWidth": "220px",
      }
    }
  >
    <p
      className="user-name"
      onClick={[Function]}
      style={
        Object {
          "color": "#616161",
          "cursor": "pointer",
          "fontSize": "24px",
          "letterSpacing": "-0.15px",
          "lineHeight": "48px",
          "marginRight": "19px",
          "textAlign": "left",
          "userSelect": "none",
        }
      }
    >
      Patricio Estrella
    </p>
    <div
      className="user-points"
      style={
        Object {
          "alignItems": "center",
          "background": "#ededed",
          "borderRadius": "100px",
          "display": "flex",
          "height": "48px",
          "justifyContent": "center",
          "userSelect": "none",
          "width": "103px",
        }
      }
    >
      <p
        className="points"
        style={
          Object {
            "color": "#616161",
            "cursor": "pointer",
            "fontSize": "24px",
            "letterSpacing": "-0.15px",
            "lineHeight": "48px",
            "textAlign": "left",
            "userSelect": "none",
          }
        }
      >
        2000
      </p>
      <img
        alt="coin"
        className="coin"
        height="24px"
        src={Object {}}
        width="24px"
      />
    </div>
  </div>
</div>
`;
